### PR TITLE
fix: proper support for it.each 

### PR DESCRIPTION
**Pull request by k-yle**
retrying #701, this time without the regression for `no-done-callback` 🙊

`no-done-callback` now works for jest-each's tagged-template syntax, but intentionally ignores it.each's table/array syntax because I have no idea how to handle [this edge case](https://github.com/jest-community/eslint-plugin-jest/issues/710#issuecomment-726081134).

### CODE REVIEW

File: src/rules/no-disabled-tests.ts [hunk 1]
```diff
@@ -39,6 +39,9 @@ export default createRule({
       CallExpression(node) {
         const functionName = getNodeName(node.callee);
 
+        // prevent duplicate warnings for it.each()()
+        if (node.callee.type === 'CallExpression') return;
```
Comment (G-Rath) [inline src/rules/no-disabled-tests.ts:43]:
```suggestion
        if (node.callee.type === 'CallExpression') {
          return;
        }
```

**Review by G-Rath — APPROVED** [review]

**Comment by SimenB** [comment]
@k-yle is this still draft?

**Comment by k-yle** [comment]
sorry, it's ready @SimenB 

I thought someone else might try to make it work for this edge case:  https://github.com/jest-community/eslint-plugin-jest/issues/710#issuecomment-726047708 but it seems to be very hard

**Comment by SimenB** [comment]
We can do that later 👍

**Review by G-Rath — APPROVED** [review]
sorry for the late review - looks good to me, lets ship it!
